include/cuco/detail/static_multimap/device_view_impl.inl

```diff
@@ -569,17 +569,19 @@ class static_multimap<Key, Value, Scope, Allocator, ProbeSequence>::device_view_
    *
    * @tparam uses_vector_load Boolean flag indicating whether vector loads are used
    * @tparam CG Cooperative Group type
+   * @tparam ProbeKey Probe key type that is convertible to the map's `key_type`
```

Comment:
It does not need to be convertible to `key_type`. The generic definition would be that `std::invoke_result<KeyEqual, ProbeKey, Key>` must be well-formed.

Reply:
`invoke_result<Hash, ProbeKey>` must also be well-formed. And if `key_equal(ProbeKey, Key)` returns true, then `hash(ProbeKey) == hash(Key)` must also be true. 

See https://www.cppstories.com/2021/heterogeneous-access-cpp20/#the-performance-gains-with-heterogeneous-lookup

Reply:
> It does not need to be convertible to key_type

Do we need `ProbeKey` being convertible to `key_type` for template `initial_slot(CG const& g, T const& k)` (required by https://github.com/NVIDIA/cuCollections/pull/174#discussion_r890385352)? My idea was to cast `T` to `key_type` when invoking the base implementation:
```cpp
  template <typename CG, typename T>
  __device__ __forceinline__ iterator initial_slot(CG const& g, T const& k) noexcept
  {
    return probe_sequence_.initial_slot(g, static_cast<Key>(k));
  }
```

> The generic definition would be that std::invoke_result<KeyEqual, ProbeKey, Key> must be well-formed

Something like this?
```cpp
static_assert(std::is_same_v<std::invoke_result_t<KeyEqual, ProbeKey, Key>, bool>);
```

> And if key_equal(ProbeKey, Key) returns true, then hash(ProbeKey) == hash(Key) must also be true.

Do we want a runtime check for this or just a statement in the doc?

Reply:
> Do we want a runtime check for this or just a statement in the doc?

That would just be a stated pre-condition in the docs.

Reply:
`std::is_invocable_v<KeyEqual, ProbeKey, Key>` is what you want.

https://en.cppreference.com/w/cpp/types/is_invocable

Reply:
Maybe `is_invocable_r_v`?
```cpp
// Determines whether Fn can be invoked with the arguments
// ArgTypes... to yield a result that is convertible to R
template <class R, class Fn, class... ArgTypes>
struct is_invocable_r;
```

Reply:
Yeah, that would be even better.